### PR TITLE
[BugFix] empty_database_backup_success (#17217)

### LINKED ISSUE #17217: empty database backup success
### Steps to reproduce the behavior (Required)
backup a empty database

### Expected behavior (Required)
fail 

### Real behavior (Required)
success

### StarRocks version (Required)
4c93b2674435f195266209709dd45ea986fc5c29

### PR DESCRIPTION
**Pull request by srlch**
## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #17217

## Problem Summary(Required) ：
Problem:
new privilege framework miss the check code for empty database backup

Solution:
add the check code

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3
  - [ ] 2.2

### CODE REVIEW

**Comment by srlch** [comment]
run starrocks_fe_unittest

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/PrivilegeCheckerV2.java [hunk 1]
```diff
@@ -1511,6 +1511,9 @@ public Void visitShowSnapshotStatement(ShowSnapshotStmt statement, ConnectContex
         public Void visitBackupStatement(BackupStmt statement, ConnectContext context) {
             checkSystemRepository(context);
             List<TableRef> tableRefs = statement.getTableRefs();
+            if (tableRefs.size() == 0) {
```
Comment (chaoyli) [inline fe/fe-core/src/main/java/com/starrocks/sql/analyzer/PrivilegeCheckerV2.java:1514]: Add the tableName in the log.

**Review by chaoyli — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/PrivilegeCheckerV2.java (hunk 1 above)
Comment (srlch) [reply fe/fe-core/src/main/java/com/starrocks/sql/analyzer/PrivilegeCheckerV2.java:1514]: There is no table in database. I have changed the error msg include the dbname

**Comment by srlch** [comment]
run label

**Review by nshangyiming — APPROVED** [review]

**Review by chaoyli — APPROVED** [review]

**Review by chaoyli — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by srlch** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 2 / 4 (50.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|0|2|00.00%|[1269, 1270]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeCheckerV2.java|2|2|100.00%|[]|

**Review by gengjun-git — APPROVED** [review]

**Pull request merged**